pkg/lib/indexer/indexer.go

```diff
@@ -258,6 +221,33 @@ func (i ImageIndexer) buildDockerfile(binarySourceImage, workingDir, tag string)
 	return nil
 }
 
+func (i ImageIndexer) getDatabaseFile(workingDir, fromIndex string) (string, error) {
+	if fromIndex == "" {
+		return path.Join(workingDir, defaultDatabaseFile), nil
```

Comment:
This is the actual fix, the rest is refactoring, correct?

Reply:
The real fix is to make sure the working directory is adjacent to the database file (lines 68 and 124) `workingDir := path.Dir(databaseFile)`. The rest is refactoring yes.

Reply:
Only refactored because I need the same thing for `filter`.